### ISSUE TITLE
[Feature request] - Nested ZIP files support

### ISSUE THREAD

**Issue by luzemario** [libxmp/libxmp#130]
Please add support for several levels deep nested ZIP files, so one can listen to http://modarchive.org torrent files without unzipping every ZIP inside ZIP files, as XMPlay for Windows does. This enhancement can be applied to both Linux XMP and Android versions.

**Comment by cmatsuoka** [comment]
We had recursive depacking a few versions ago, but we had problems related to deleting closed descriptors in Windows (Dominik fixed that), and after that we had problems with the read-from-memory I/O wrappers. So the current version dropped this support because "no one used it". I think we need a more sophisticated way to handle packed files.

After the Oxidrizzle experiment in Rust, I guess the "next-gen" xmp will be implemented in C++ -- it allows most of the advantages from the Rust version (minus the bulletproof memory management), and adds a few more things that would be interesting, like using third-party format players without having to port them first to a different language. In this case, depacking support will be entirely handled outside the player library, so we can do whatever is needed without disturbing the player code.

**Comment by luzemario** [comment]
Such feature is very wanted mainly on Android, since on Linux some sort of shell script can do the job. It appears to be that the issues you described are Windows-only related, but anyway, if depacking will be handled entirely outside, you can close this at your own discretion.

Thanks for the quick reply!

**Comment by cmatsuoka** [comment]
One of the reasons to handle depacking outside the library is to allow an Android application to browse inside zipfiles and select a number of files to play. This is not possible with the current built-in depacker.

**Comment by luzemario** [comment]
You're right. It's clear to me. So an app, i.e. a file manager or zip decompressor can send a list of files to be played one by one by the Android app.

I could not find a better place to fill up this request, so you can leave this bug open until the new version comes out, or close this since it is not a bug against the xmp library itself. Feel free to chose what you want to do with it.

Thanks for the explanation.

**Comment by NoSuck** [comment]
> ... since on Linux some sort of shell script can do the job.

Or better yet, use [archivemount](https://www.cybernoia.de/software/archivemount.html).  Thank you, by the way, for seeding the Mod Archive's torrents (presumably).
